lms/static/js/spec/student_account/account_settings_factory_spec.js

```diff
@@ -148,4 +148,158 @@ define(['backbone',
+                Helpers.expectSettingsSectionsAndFieldsToBeRenderedWithMessage(accountSettingsView);
+            });
+
+            it('expects all fields to behave correctly for enterprise learners', function() {
+                var accountSettingsView, i, view, sectionsData, textFields, dropdownFields;
+                requests = AjaxHelpers.requests(this);
+
+                accountSettingsView = createEnterpriseLearnerAccountSettingsPage();
+
+                AjaxHelpers.respondWithJson(requests, Helpers.createAccountSettingsData());
+                AjaxHelpers.respondWithJson(requests, Helpers.TIME_ZONE_RESPONSE);
+                AjaxHelpers.respondWithJson(requests, Helpers.createUserPreferencesData());
+                AjaxHelpers.respondWithJson(requests, {});  // Page viewed analytics event
+
+                sectionsData = accountSettingsView.options.tabSections.aboutTabSections;
+
+                expect(sectionsData[0].fields.length).toBe(7);
+
+                // Verify that username, name and email fields are readonly
+                textFields = [sectionsData[0].fields[0], sectionsData[0].fields[1], sectionsData[0].fields[2]];
+                for (i = 0; i < textFields.length; i++) {
+                    view = textFields[i].view;
+
+                    FieldViewsSpecHelpers.verifyReadonlyTextField(view, {
+                        title: view.options.title,
+                        valueAttribute: view.options.valueAttribute,
+                        helpMessage: view.options.helpMessage,
+                        validValue: 'My Name',
+                        defaultValue: ''
+                    }, requests);
+                }
+
+                // Verify un-editable country dropdown field
+                view = sectionsData[0].fields[5].view;
+                FieldViewsSpecHelpers.verifyReadonlyDropDownField(view, {
+                    title: view.options.title,
+                    valueAttribute: view.options.valueAttribute,
+                    helpMessage: '',
+                    validValue: Helpers.FIELD_OPTIONS[1][0],
+                    editable: 'never',
+                    defaultValue: null
+                });
+
+                expect(sectionsData[1].fields.length).toBe(4);
+                dropdownFields = [
+                    sectionsData[1].fields[0],
+                    sectionsData[1].fields[1],
+                    sectionsData[1].fields[2]
+                ];
+                _.each(dropdownFields, function(field) {
+                    view = field.view;
+                    FieldViewsSpecHelpers.verifyDropDownField(view, {
+                        title: view.options.title,
+                        valueAttribute: view.options.valueAttribute,
+                        helpMessage: '',
+                        validValue: Helpers.FIELD_OPTIONS[1][0],
+                        invalidValue1: Helpers.FIELD_OPTIONS[2][0],
+                        invalidValue2: Helpers.FIELD_OPTIONS[3][0],
```

Comment:
difficult to understand the values behind `[1][0], [2][0] ... ` etc etc. any other way or comments would be helpful